### ISSUE TITLE
Todo

### ISSUE THREAD

**Issue by WenheLI** [imgcook/datacook#2]
This will be a thread for discussion about what we should cover in the future.

IMO, we should have four major aspects:

- Tabular
- Text
- Image
- Sound

And we should implement some common features like `dataset splitting`, `shuffle`, `data augmentation`,  `visualizing?`

Besides, data-type specific operations are also needed:

- Generic
   - Train test split
   - Random sampler
   - Scalers (Standard, Robust, MinMax)
- Image 
   - Basic I/O
   - Filter
   - Crop
   - Resize
   - Rotating
   - Augmentation
   - color_mode
   - Normalize
   - ([Some more ideas](https://keras.io/api/preprocessing/image/#image-data-preprocessing))
- Text
   - Embedding
   - Serialize
   - stats analysis
   - Tokenization
   - Stemming
   - Stopword removal
   - Padding
- Soung
   - FFT?
- Tabular Data
  - Encodings 
  - I/O

Discussions are welcome here!

**Comment by risenW** [comment]
> > > @risenW Just update the codebase with `ts`. If this is okay with you, shall we turn this to public?
> > 
> > 
> > Looks great! Yea we can make it public. Quick question, are we using Typescript or Vanilla JS?
> 
> Definitely TypeScript. It makes life easier!

Okay, noted 😁

**Comment by yorkie** [comment]
Just removed because this is stale and almost done.
